pact/dao/dao.repl

```diff
@@ -0,0 +1,171 @@
+(begin-tx)
+(env-data
+ { 'ns-admin-keyset: ["admin"]
+ , 'ns-operate-keyset: ["operate"]
+ , 'ns-genesis-keyset: { "keys": [], "pred": "="} })
+(load "../root/fungible-v2.pact")
+(load "../root/coin.pact")
+(load "../root/ns.pact")
+
+(env-data
+  { 'dao-ns-user: ["dao-user"]
+  , 'dao-ns-admin: ["dao-admin"]
+  , 'dao-ns: "dao"
+  , 'upgrade: false
+  })
+(env-keys ["operate", "dao-user", "dao-admin"])
+
+(load "dao-ns.pact")
+
+(env-data { 'dao-ns: "dao" })
+(env-chain-data
+  {"block-height": 1675409
+  ,"block-time": (time "2021-06-01T12:00:00Z")
+  ,"chain-id": "0"
+  ,"gas-limit": 10000000
+  ,"gas-price": 0.0
+  ,"prev-block-hash": ""
+  ,"sender": ""})
+(load "dao.pact")
+(typecheck "dao.dao-v1" true)
+
+; Check init State
+(coin.details DAO_ACCT_NAME)
+(use dao.dao-v1)
+(view-state)
+(view-guardians)
+(view-ambassadors)
+(is-dao-frozen)
+(expect "state hash checkpoint" "qp44Pd5Mzox-D2YiGakiC-zTzLeyjrFuyXXAnyR3duU" (state-hash))
+(commit-tx)
+
+(begin-tx)
+(load "tests/populate-users.repl")
+(use dao.dao-v1)
+
+; -----------------------------------------------------------------------------
+; basic guardian functionality tests
+; -----------------------------------------------------------------------------
+(env-sigs [
+  { 'key: 'guardian0
+  , 'caps:
+      [(coin.TRANSFER 'guardian0 DAO_ACCT_NAME GUARDIAN_KDA_REQUIRED)]}
+  { 'key: 'guardian1
+  , 'caps:
+      [(coin.TRANSFER 'guardian1 DAO_ACCT_NAME GUARDIAN_KDA_REQUIRED)]}
+  { 'key: 'guardian2
+  , 'caps:
+      [(coin.TRANSFER 'guardian2 DAO_ACCT_NAME GUARDIAN_KDA_REQUIRED)]}
+  ])
+(register-guardian 'guardian0 (read-keyset 'guardian0))
+(register-guardian 'guardian1 (read-keyset 'guardian1))
```

Comment:
Good chance here to test for failure of various ops for insufficient guardian count

Reply:
Good call, I didn't notice I was going with a 50% vote instead of a > 50% vote. That said, a solo guardian is supposed to have the authority to do whatever. There's a 1 day cooldown between proposal and upgrade, so I'm not too worried about someone swoping in.